### DISCUSSION TITLE
Possible to test through an address/port change?

### DISCUSSION THREAD

**Discussion by jg3** [esnet/iperf#1478] [Q&A]
I have a system that re-directs (UDP) traffic based on a ruleset.  I want to test throughput / packet loss through that system.  iPerf looks great for that but if I configure the client to send to address .1 port 1111 but the server is on address .100 port 9999 the client either hangs or gives up with `iperf3: error - unable to send control message: Bad file descriptor`

Is there a way to do what I'm trying to do here?  Maybe set a seprate control ip/port?

**Comment by bmah888** [comment]
Can you post the command-line invocations of iperf3 for both the client and server? It'd help to know the exact arguments you are using. Off-hand I can think of two possible causes for this behavior.

1. iperf3 uses the `-p` port number for both the control connection and the various data streams. If you're translating the UDP port numbers for the test treams, but not for the control connection (which runs over TCP), I can imagine packets winding up on the "wrong" port number on the server side.
2. Related to this, is the control connection even working? You can generally get some idea of this by watching the server side output when the client attempts its connection.

Bruce.
